streams/src/main/java/org/apache/kafka/streams/processor/internals/SinkNode.java

```diff
@@ -97,6 +96,10 @@ public void process(final K key, final V value) {
                                     keyClass,
                                     valueClass),
                     e);
+        } catch (final StreamsException e) {
```

Comment:
Also as we have discussed offline, I think it is more valuable to log an error saying that `Aborting to send this record because a previous sent record returned an error` and include the following information: error message, key / value / topic / offset of the previous sent record that triggered the exception, key / value of the aborted record.